File: 1431. Kids With the Greatest Number of Candies.js
// https://leetcode.com/problems/kids-with-the-greatest-number-of-candies/description

/**
 * @param {number[]} candies
 * @param {number} extraCandies
 * @return {boolean[]}
 */
var kidsWithCandies = function (candies, extraCandies) {
  let maxCandy = 0;

  candies.map((candy) => {
    if (candy > maxCandy) maxCandy = candy;
    return candy + extraCandies;
  });

  return candies.map((candy) => candy >= maxCandy);
};

File: 1431. Kids With the Greatest Number of Candies.py
# https://leetcode.com/problems/kids-with-the-greatest-number-of-candies/description

class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        maxCandy = max(candies)
        return list(map(lambda candy: candy + extraCandies >= maxCandy, candies))
        